packages/eslint-plugin/src/rules/no-misused-spread.ts

```diff
@@ -0,0 +1,255 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+import * as tsutils from 'ts-api-utils';
+import * as ts from 'typescript';
+
+import {
+  createRule,
+  getConstrainedTypeAtLocation,
+  getParserServices,
+  isBuiltinSymbolLike,
+  isPromiseLike,
+  isTypeFlagSet,
+} from '../util';
+
+type Options = [
+  {
+    allowClassInstances: boolean;
+  },
+];
+
+type MessageIds =
+  | 'noStringSpreadInArray'
+  | 'noSpreadInObject'
+  | 'noFunctionSpreadInObject'
+  | 'noClassSpreadInObject';
+
+export default createRule<Options, MessageIds>({
+  name: 'no-misused-spread',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Disallow using the spread operator when it might cause unexpected behavior',
+      recommended: 'strict',
+      requiresTypeChecking: true,
+    },
+    messages: {
+      noStringSpreadInArray:
+        "Using the spread operator on a string can cause unexpected behavior. Prefer `String.split('')` instead.",
+
+      noSpreadInObject:
+        'Using the spread operator on `{{type}}` can cause unexpected behavior.',
```

Comment:
[Docs] I think we'll need to be precise about when the spread operators are or aren't allowed. Spreading an Iterable in an array is intentionally allowed.

```suggestion
        'Using the spread operator on `{{type}}` in an object can cause unexpected behavior.',
```

Reply:
Done